scipy/stats/_multicomp.py

```diff
@@ -234,33 +296,63 @@ def dunnett(
             stats.ttest_ind(
                 obs_, control, alternative=alternative, random_state=rng
             ).statistic
-            for obs_ in observations
+            for obs_ in samples
         ])
 
     pvalue = pvalue_dunnett(
         rho=rho, df=df,
-        statistic=statistic, alternative=alternative,
+        statistic=abs(statistic), alternative=alternative,
         rng=rng
     )
 
     control_mean = np.mean(control)
-    observations_mean = np.array([np.mean(obs_) for obs_ in observations])
+    observations_mean = np.array([np.mean(obs_) for obs_ in samples])
     std = np.sqrt((
         np.sum(control**2)
-        + np.sum([obs_**2 for group in observations for obs_ in group])
+        + np.sum([obs_**2 for group in samples for obs_ in group])
         - n_group*(control_mean**2 + np.sum(observations_mean**2))
     ) / df)
 
     return DunnettResult(
         statistic=statistic, pvalue=pvalue,
+        _alternative=alternative,
         _rho=rho, _df=df, _std=std,
         _observations_mean=observations_mean,
         _control_mean=control_mean
     )
 
 
 def iv_dunnett(
-    observations: npt.ArrayLike, control: npt.ArrayLike
+    samples: npt.ArrayLike, control: npt.ArrayLike, random_state: SeedType
+) -> tuple[npt.ArrayLike, np.ndarray, SeedType]:
+    """Input validation for Dunnett's test."""
+    rng = check_random_state(random_state)
+
+    ndim_msg = "Control and samples groups must be 1D arrays"
+    n_obs_msg = "Control and samples groups must have at least 1 observation"
+
+    # control checks
+    control = np.asarray(control)
+    if control.ndim > 1:
+        raise ValueError(ndim_msg)
+
+    if len(control) < 1:
+        raise ValueError(n_obs_msg)
+
+    # samples checks
+    for sample in samples:
+        sample = np.asarray(sample)
+        if sample.ndim > 1:
+            raise ValueError(ndim_msg)
+
+        if len(sample) < 1:
+            raise ValueError(n_obs_msg)
+
+    return samples, control, rng
```

Comment:
`samples` isn't modified.

Reply:
You expect it to be a list of np.array?

Reply:
I thought that was the intent, no? That way, if the samples are passed as lists, they do not need to re-converted to arrays later.